data-mongodb/src/main/java/io/micronaut/data/mongodb/transaction/TransactionalClientSessionInterceptor.java

```diff
@@ -30,19 +30,19 @@
  * @author Denis Stepanov
  * @since 3.3
  */
-@Requires(classes = ClientSession.class, beans = MongoSynchronousTransactionManager.class)
+@Requires(classes = ClientSession.class, beans = MongoSynchronousTransactionManagerImpl.class)
 @Prototype
 public final class TransactionalClientSessionInterceptor implements MethodInterceptor<ClientSession, Object> {
 
-    private final MongoSynchronousTransactionManager transactionManager;
+    private final MongoSynchronousTransactionManagerImpl transactionManager;
 
     /**
      * Default constructor.
      *
      * @param transactionManager The transactionManager
      */
     @Internal
-    TransactionalClientSessionInterceptor(MongoSynchronousTransactionManager transactionManager) {
+    TransactionalClientSessionInterceptor(MongoSynchronousTransactionManagerImpl transactionManager) {
```

Comment:
Just curious, why is `MongoSynchronousTransactionManagerImpl` use here (and on some other places) instead of interface `MongoSynchronousTransactionManager` ?

Reply:
It accesses the current client session; not sure if it's something we want to expose ATM.